statter_test.go

```diff
@@ -0,0 +1,417 @@
+package statter_test
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hamba/statter"
+	"github.com/hamba/statter/tags"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestNew_HandlesOptions(t *testing.T) {
+	m := &mockSimpleReporter{}
+	m.On("Counter", "test-test", int64(2), [][2]string{{"tag", "test"}})
+
+	stats := statter.New(m, time.Second, statter.WithSeparator("-"))
+
+	stats.With("test").Counter("test", tags.Str("tag", "test")).Inc(2)
+
+	err := stats.Close()
+	assert.NoError(t, err)
+
+	m.AssertExpectations(t)
+}
+
+func TestStatter_CallsReporter(t *testing.T) {
+	m := newWaitingReporter()
+	m.On("Counter", "test", int64(2), [][2]string{{"tag", "test"}})
+
+	stats := statter.New(m, time.Millisecond)
+	t.Cleanup(func() { _ = stats.Close() })
+
+	stats.Counter("test", tags.Str("tag", "test")).Inc(2)
+
+	select {
+	case <-time.After(time.Second):
+		assert.FailNow(t, "expected call to reporter timed out")
+	case <-m.Ch():
+	}
+}
```

Comment:
Missing assertExpectations to verify that `reporter.Counter` was called as expected